src/shell.cpp

```diff
@@ -0,0 +1,243 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <cstring>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <sstream>
+#include <fstream>
+#include <map>
+
+// Environment variables
+std::map<std::string, std::string> environmentVariables;
+
+/**
+ * Splits a string into tokens based on a delimiter.
+ * @param str The string to split.
+ * @param delimiter The delimiter character.
+ * @return A vector of tokens.
+ */
+std::vector<std::string> Split(const std::string& str, char delimiter) {
+    std::vector<std::string> tokens;
+    std::istringstream tokenStream(str);
+    std::string token;
+    while (std::getline(tokenStream, token, delimiter)) {
+        tokens.push_back(std::move(token));
+    }
+    return tokens;
+}
+
+/**
+ * Executes the child process.
+ * @param command The command to execute.
+ * @param argv The arguments for the command.
+ */
+void ExecuteChildProcess(const std::string& command, char** argv) {
+    execvp(command.c_str(), argv);
```

Comment:
`execvp` finds the executable file for you, but you were asked to implement that by yourself